CRM/Batch/BAO/Batch.php

```diff
@@ -648,10 +648,11 @@ public static function getBatchFinancialItems($entityID, $returnValues, $notPres
     }
 
     $from = "civicrm_financial_trxn
-LEFT JOIN civicrm_entity_financial_trxn ON civicrm_entity_financial_trxn.financial_trxn_id = civicrm_financial_trxn.id
+INNER JOIN civicrm_entity_financial_trxn ON civicrm_entity_financial_trxn.financial_trxn_id = civicrm_financial_trxn.id
+INNER JOIN civicrm_contribution ON (civicrm_contribution.id = civicrm_entity_financial_trxn.entity_id
```

Comment:
So, how does that tie in with 

"This is relevant as the form is intended to allow financial data from other types of financial information as well eg grants, general journal entries. While these have entries in the civicrm_financial_trxn table, they don't have related civicrm_contributions, and a currency is still required."

I feel like that INNER JOIN would exclude them from being pulled in. I read that as a bad thing - but perhaps it is a good thing?